test/registry/FHIRRegistryClient.test.ts

```diff
@@ -121,78 +92,149 @@ describe('FHIRRegistryClient', () => {
+          'Attempting to download hl7.terminology.r4#1.1.2 from https://packages.fhir.org/hl7.terminology.r4/1.1.2'
         );
-        expect(latest).rejects.toThrow(Error);
-        expect(latest).rejects.toThrow(
-          'Failed to download hl7.terminology.r4#7.7.7 from https://packages.fhir.org/hl7.terminology.r4/7.7.7'
+        await expect(latest).rejects.toThrow(Error);
+        await expect(latest).rejects.toThrow(
+          'Failed to download hl7.terminology.r4#1.1.2 from https://packages.fhir.org/hl7.terminology.r4/1.1.2'
         );
       });
 
       it('should throw error when trying to get the version of a package on the packages server but returns no data', async () => {
-        const latest = client.download('hl7.terminology.r4', '7.7.9');
+        // Note: don't know of a scenario where this would occur but testing for completeness.
+        const latest = client.download('hl7.terminology.r4', '1.1.1');
         expect(loggerSpy.getLastMessage('info')).toBe(
-          'Attempting to download hl7.terminology.r4#7.7.9 from https://packages.fhir.org/hl7.terminology.r4/7.7.9'
+          'Attempting to download hl7.terminology.r4#1.1.1 from https://packages.fhir.org/hl7.terminology.r4/1.1.1'
         );
-        expect(latest).rejects.toThrow(Error);
-        expect(latest).rejects.toThrow(
-          'Failed to download hl7.terminology.r4#7.7.9 from https://packages.fhir.org/hl7.terminology.r4/7.7.9'
+        await expect(latest).rejects.toThrow(Error);
+        await expect(latest).rejects.toThrow(
+          'Failed to download hl7.terminology.r4#1.1.1 from https://packages.fhir.org/hl7.terminology.r4/1.1.1'
         );
       });
     });
 
     describe('#downloadSpecificVersion', () => {
-      beforeEach(() => {
-        loggerSpy.reset();
-      });
-      beforeAll(() => {
-        axiosSpy = jest.spyOn(axios, 'get').mockImplementation((uri: string): any => {
-          if (
-            uri === 'https://custom-registry.example.org/hl7.terminology.r4' ||
-            uri === 'https://packages.fhir.org/hl7.terminology.r4'
-          ) {
-            return { data: TERM_PKG_RESPONSE };
-          } else if (uri === 'https://packages.fhir.org/hl7.no.latest') {
-            return {
-              data: {
-                name: 'hl7.no.latest',
-                'dist-tags': {
-                  v1: '1.5.1',
-                  v2: '2.1.1'
-                }
-              }
-            };
-          } else if (uri === 'https://packages.fhir.org/hl7.terminology.r4/3.3.3') {
-            return {
-              status: 200,
-              data: Readable.from(['3.3.3-test-data'])
-            };
-          } else if (uri === 'https://packages.fhir.org/hl7.terminology.r4/8.8.8') {
-            return {
-              status: 400,
-              data: Readable.from(['8.8.8-test-data'])
-            };
-          } else {
-            throw new Error('Not found');
-          }
+      describe('#CreateURL', () => {
```

Comment:
Ah... Looking at the NPMRegistryClient tests, I think I understand why this is here. But I still don't think it's needed -- because the NPM client has separate logic for when there is a tarball URL in the manifest and when there isn't, but the FHIR client works a bit different so I don't think the distinction is there.